planner/core/logical_plan_builder.go

```diff
@@ -2611,6 +2614,12 @@ func (b *PlanBuilder) buildUpdateLists(tableList []*ast.TableName, list []*ast.A
 		col := assign.Col
 
 		dbName := col.DBName.L
+		// To solve issue#10028, we need to get database name by the table alias name.
+		for _, tbl := range tableList {
```

Comment:
will it be faster if we pre-build a `map[tblName]DBName` and save this `for` loop?